Sources/MongoSwift/BSON/Document.swift

```diff
@@ -76,10 +76,15 @@ public struct Document: ExpressibleByDictionaryLiteral, ExpressibleByArrayLitera
      *
      * - Returns: a new `Document`
      */
-    public init(dictionaryLiteral doc: (String, BsonValue?)...) {
+    public init(dictionaryLiteral keyValuePairs: (String, BsonValue?)...) {
+        // make sure all keys are unique
+        if Set(keyValuePairs.map { $0.0 }).count != keyValuePairs.count {
```

Comment:
this is the same behavior as `Dictionary`:
```
9> let d = ["a": 1, "a": 2]
Fatal error: Dictionary literal contains duplicate keys
```

Reply:
If we're mirroring Dictionary here, I don't think we need to make this behavior something users can turn on/off, per your earlier [comment](https://github.com/mongodb/mongo-swift-driver/pull/111#issuecomment-414538230):

> another question here is whether this behavior should be a setting users can turn on and off, and if so, when would be the appropriate time for them to specify that?

I can't think of a valid use case for creating BSON documents with duplicate keys, and most other drivers that use native types for document representations (thinking Python, PHP, Node) wouldn't support this at all.